rewriter/src/main/java/com/facebook/coresql/rewriter/OrderByRewriter.java

```diff
@@ -73,23 +67,26 @@ public void visit(OrderByClause node, Void data)
 
     private static class OrderByPatternMatcher
             extends SqlParserDefaultVisitor
-            implements PatternMatcher<Set<AstNode>>
     {
-        private final Set<AstNode> matchedNodes;
+        private final ImmutableSet.Builder<AstNode> builder;
         private int depth;
         private static final int MINIMUM_SUBQUERY_DEPTH = 2; // Past this depth, all queries we encounter are subqueries
 
         public OrderByPatternMatcher()
         {
-            this.matchedNodes = new HashSet<>();
+            this.builder = ImmutableSet.builder();
+        }
+
+        public static Set<AstNode> getMatchedNodes(AstNode root)
```

Comment:
The same problem with the outer class apply here as well. I can call `getMatchedNodes(sqlRoot1); getMatchedNodes(sqlRoot2);` and the second call would actually return me matched nodes from both sqls. It's slightly less bad here because it's a private class, but still a bad pattern. cc @kaikalur

Reply:
Changed the constructor to take in the root as a final member variable and matchPattern() to be parameter-less.

Done.